# HG changeset patch
# User Travis Shirk <travis@pobox.com>
# Date 1366683753 21600
# Branch stable
# Node ID bfbfc2c5709c4084ecaeed3410600d94d2a21a0d
# Parent  e2ebd8a246ecca8340bbc4ac858c7a946524571f
Fix for setting of non standard text frames. Fixes #26

diff -r e2ebd8a246ecca8340bbc4ac858c7a946524571f -r bfbfc2c5709c4084ecaeed3410600d94d2a21a0d src/eyed3/id3/frames.py
--- a/src/eyed3/id3/frames.py	Wed Mar 13 21:28:01 2013 -0600
+++ b/src/eyed3/id3/frames.py	Mon Apr 22 20:22:33 2013 -0600
@@ -1364,7 +1364,8 @@
         the same Id is already in the list it's value is changed, otherwise
         the frame is added.
         '''
-        assert(fid[0] == "T" and fid in list(ID3_FRAMES.keys()))
+        assert(fid[0] == "T" and (fid in ID3_FRAMES or
+                                  fid in NONSTANDARD_ID3_FRAMES))
 
         if fid in self:
             self[fid][0].text = text
